presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -95,6 +100,11 @@ public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
             if (simplified.equals(BooleanLiteral.TRUE_LITERAL)) {
                 return source;
             }
+            // TODO: this needs to check whether the boolean expression coerces to false in a more general way.
+            // E.g., simplify() not always produces a literal when the expression is constant
+            else if (BooleanLiteral.FALSE_LITERAL.equals(simplified) || simplified instanceof NullLiteral) {
```

Comment:
Yes, we generally write it this way if the thing being compared can be null. In this case, it's not necessary, since `simplified` can't be null. It ended up like that because I just copied that code from somewhere else. I'll update it.